src/main/java/billgates/view/gui/RegisterDialog.java

```diff
@@ -0,0 +1,111 @@
+
+        // Set the root panel and its layout
+        JPanel root = new JPanel();
+        this.setContentPane(root);
+        root.setLayout(new BorderLayout());
+
+        // Components
+        if (true){
+            JPanel panel = new JPanel();
+            root.add(panel, BorderLayout.CENTER);
+
+            JLabel nameLabel = new JLabel("<HTML><font color=black>Name</font><font color=red>*</font>" +
+                    "<font color=black>:</font></html>");
+            JLabel nickLabel = new JLabel("Nickname:");
+            JLabel passwordLabel = new JLabel("<HTML><font color=black>Password</font><font color=red>*</font>" +
+                    "<font color=black>:</font></html>");
+            JLabel confirmPasswordLabel = new JLabel("<HTML><font color=black>Confirm password</font>" +
+                    "<font color=red>*</font><font color=black>:</font></html>");
+            JLabel emailLabel = new JLabel("Email:");
+            panel.add(nameLabel);
+            panel.add(nameInput);
+            panel.add(nickLabel);
+            panel.add(nickInput);
+            panel.add(passwordLabel);
+            panel.add(passwordInput);
+            panel.add(confirmPasswordLabel);
+            panel.add(confirmPasswordInput);
+            panel.add(emailLabel);
+            panel.add(emailInput);
+
+            passwordInput.setEchoChar('*');
+            confirmPasswordInput.setEchoChar('*');
+        }
+
+        if (true){
+            JPanel panel = new JPanel();
+            root.add(panel, BorderLayout.SOUTH);
+
+            panel.add(cancelButton);
+            panel.add(createButton);
+        }
+
+        // Case 1: Click cancelButton
+        cancelButton.addActionListener(e -> {
+            setVisible(false);
+        });
+
+        // Case 2: Click createButton
+        createButton.addActionListener((e -> {
+            if (nameInput.getText().equals("") | passwordInput.getPassword().length == 0 |
+                    confirmPasswordInput.getPassword().length == 0){
+                JOptionPane.showMessageDialog(null,
+                        "Blank field detected ! Please fill in the field with * !", "Warning",
+                        JOptionPane.WARNING_MESSAGE);
+            } else if (!matchPassword()) {
+                JOptionPane.showMessageDialog(null, "Passwords don't match!",
+                        "Error", JOptionPane.ERROR_MESSAGE);
+            } else {
+                accepted = true;
+                setVisible(false);
+            }
+        }));
+    }
+
+    public String getInput(){
+        return String.format("Name: %s\n Nickname: %s\n Password: %s\n Email: ", nameInput.getText(),
+                nickInput.getText(), Arrays.toString(passwordInput.getPassword()), emailInput.getText());
+    }
+
+    // Execute
+    public boolean exec(){
+        this.setTitle("Register");
+        this.setModal(true);
+        this.setSize(300, 220);
+        this.setResizable(false);
```

Comment:
![image](https://user-images.githubusercontent.com/8016180/201282025-8f12d3cb-e50d-4f38-a063-d069094d1323.png)
I'm not sure if it's my display or something, but the register dialogue doesn't look like what it is intended to look like?

Reply:
I don't know what happens here, but it works for Scott and me. Like this:
![image](https://user-images.githubusercontent.com/60980117/201430501-559ff0ca-2e7e-4328-a142-a310d7b0cb22.png)
Maybe it doesn't work for mac?
I will try to figure it out.